### ISSUE TITLE
PolyNull Object as receiver error

### ISSUE THREAD

**Issue by Ao-senXiong** [eisop/checker-framework#532]
In this code snippet, the expected error is `dereferenc of possibly-null reference nbl`. However, a `method invocation invalid` error is also raised.

```
import org.checkerframework.checker.nullness.qual.PolyNull;

class Demo {
    void foo(@PolyNull Object nbl) {
        if (nbl == null) {
            nbl.toString();
        }
    }
}
```

Error message is listing for reference:
```
Demo.java:6: error: [dereference.of.nullable] dereference of possibly-null reference nbl
            nbl.toString();
            ^
Demo.java:6: error: [method.invocation.invalid] call to toString() not allowed on the given receiver.
            nbl.toString();
                        ^
  found   : @FBCBottom @PolyNull Object
  required: @Initialized @NonNull Object

```

Another code snippet with `@Nullable` annotation is 
```
import org.checkerframework.checker.nullness.qual.Nullable;

class Demo {
    void foo(@Nullable Object nbl) {
        if (nbl == null) {
            nbl.toString();
        }
    }
}

```

And the corresponding error message are listing below
```
Demo.java:6: error: [dereference.of.nullable] dereference of possibly-null reference nbl
            nbl.toString();
            ^
1 error

```

**Comment by wmdietl** [comment]
This came up here: https://github.com/eisop/checker-framework/pull/525#discussion_r1246813988 . Try to keep references between different issues and PRs, to make the history easier to follow.

It's noteworthy here that a different error is issued for `@Nullable` instead of `@PolyNull` in this test case.

**Comment by Ao-senXiong** [comment]
> This came up here: [#525 (comment)](https://github.com/eisop/checker-framework/pull/525#discussion_r1246813988) . Try to keep references between different issues and PRs, to make the history easier to follow.
> 
> It's noteworthy here that a different error is issued for `@Nullable` instead of `@PolyNull` in this test case.

Thanks, already added.

**Comment by Ao-senXiong** [comment]
For a simpler case without dataflow refinement, I think we are still expecting to dereference possible-null reference errors because of soundness.

```
import org.checkerframework.checker.nullness.qual.PolyNull;

class Demo {
    void foo(@PolyNull Object nbl) {
            nbl.toString();
    }
}
```

**Comment by wmdietl** [comment]
Yes, directly dereferencing a `PolyNull` reference should give you an error, like for a `Nullable ` reference.
